web/src/settings_profile_fields.js

```diff
@@ -274,16 +274,11 @@ function add_choice_row(e) {
     create_choice_row(choices_div);
 }
 
-function delete_choice_row(e) {
-    const $row = $(e.currentTarget).parent();
+function delete_choice_row() {
+    const $row = $(this).parent();
     $row.remove();
 }
 
-function delete_choice_row_for_edit(e, $profile_field_form, field) {
-    delete_choice_row(e);
-    disable_submit_btn_if_no_property_changed($profile_field_form, field);
-}
```

Comment:
Why remove this function, I think it was good for readability.
If that was for typescript, you can just skip passing `e` argument here and directly pass `this` as argument since that's what we are using `e` for anyways.
Something like this:
```ts
function delete_choice_row_for_edit(
    row: HTMLElement,
    $profile_field_form: JQuery,
    field: CustomProfileField,
): void {
    delete_choice_row(row);
    disable_submit_btn_if_no_property_changed($profile_field_form, field);
}
```
And then just call this like this in event handler:
```ts
  .on("click", "button.delete-choice", function (this: HTMLElement) {
      delete_choice_row_for_edit(this, $profile_field_form, field);
  });
```

Reply:
Fixed.